### DISCUSSION TITLE
Installer should stop the service before update

### DISCUSSION THREAD

**Discussion by c-michail** [rustdesk/rustdesk#10038] [General]
Hello,
  Using MSI installer it should stop the service and wait the service file not to be used in order to avoid rebooting.

Thank you

**Comment by fufesou** [comment]
Hello, this is a prompt from msi. Actually, you don't need to restart. When msi is installed, the service will be stopped.
I have looked for a solution, but I haven't found a suitable one yet. The service stop operation needs to be executed earlier.
And WiX(the msi builder tool) limits the timing of this execution.

**Reply by c-michail** [reply]
Hello,
  There various arictles on how to do that:

https://www.google.com/search?q=WiX+stop+service+site:stackoverflow.com&client=firefox-b-d&sca_esv=473d468abdc3ab19&sxsrf=AHTn8zqt4oHzbAP-f7SqGDQN9mDboztxAw:1745578949253&sa=X&ved=2ahUKEwjrj8qphPOMAxUfKvsDHQJqBTYQrQIoAnoECCAQAw&biw=1920&bih=948&dpr=2

**Reply by fufesou** [reply]
Thanks for your help. We do have the `Action` to stop and delete services. However, our msi is a little different. We don't use the built-in `ServiceControl` in WiX, but a custom action.
Maybe we can split the custom action into several steps, and use the `ServiceControl`. After I finish the more urgent works, I will see how to deal with this.
